pyctuator/impl/spring_boot_admin_registration.py

```diff
@@ -3,25 +3,39 @@
 import logging
 import threading
 import urllib.parse
+from base64 import b64encode
+from dataclasses import dataclass
 from datetime import datetime
 
 from http.client import HTTPConnection
-from typing import Optional
+from typing import Optional, Dict
+
+
+@dataclass
+class RegistrationEndpoint:
+    registration_url: str
```

Comment:
What's the benefit of putting the url and authentication details in the same class? 
I expected this class to be named `Auth` and be empty, and then have a `BasicAuth` subclass, etc.

The url shouldn't be tied to the auth IMO

Reply:
I think of it as "how to connect to an endpoint" and than there is "no-auth" and "basic-auth" and maybe "oauth".
The benefit is that it reduces the number of arguments to the method and emphasis the connection between the parameters.

Reply:
but you ended up adding _more_ arguments to the pyctuator interface (`username` and `password`), which are only relevant for some types of authentication.

I really think you should replace these two with an `auth` (or `sba_auth`) argument at the pyctuator level